### DISCUSSION TITLE
How do you properly sort by relation?

### DISCUSSION THREAD

**Discussion by imtrent** [prisma/prisma#9918] [Q&A] [answered]
The documentation says this for [sorting by a relation](https://www.prisma.io/docs/concepts/components/prisma-client/filtering-and-sorting#sort-by-relation):

> Sort by relation
   You can also sort by properties of a relation. For example, the following query sorts all posts by the author's email address:
```
const posts = await prisma.post.findMany({
  orderBy: {
    author: {
      email: 'asc',
    },
  },
})
```

When I try to sort by the relation, it only gives me access to sort by `_count`. What could I be doing wrong when sorting based on properties within the relation -- like how the documentation is doing it.

Here are the prisma models I'm working with
```prisma
model Post {
  id String @id @default(uuid())
  title String
  content String
  createdAt DateTime @default(now())
  updatedAt DateTime @default(now()) @updatedAt
  activity UserPostActivity[]
}

model UserPostActivity {
  id String   @id @default(uuid())
  userId String
  postId String
  lastViewedAt DateTime @default(now())
  user User @relation(fields: [userId], references: [id], onDelete: Cascade)
  post Post @relation(fields: [postId], references: [id], onDelete: Cascade)
}
```

Here is the query I have that says I can only sort by `_count`
```typescript
const diagrams = await this.prismaService.post.findMany({
  orderBy: {
    activity: {
      lastViewedAt: 'desc',
    },
  },
});
```

**Comment by TasinIshmam** [comment, marked as answer]
Hey @imtrent :wave: 

Are you trying to find a bunch of posts, include the connected `activity` records and order the `activity` records based on the `lastViewedAt` date? 

Here is the syntax: 

```typescript
  const data = await prisma.post.findMany({

    include: {
      activity: {
        orderBy: {
          lastViewedAt: 'desc'
        }
      }
    }
  });
```

The syntax you're trying to use in the question only allows for `_count` because it's the ordering criteria for the `post` records...not the included `activity` records. This makes sense because the only way to sort `post` with some property of `activity` is with aggregate properties of `activity` (such as count) as the relationship is `one-to-many`.  

By the way, you're trying to access a `diagram` model with the `prismaservice`. I assume this will be `post` instead? 

Let me know if you have any questions!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
